miio/devicestatus.py

```diff
@@ -235,35 +230,32 @@ def setting(
     """
 
     def decorator_setting(func):
-        property_name = func.__name__
+        property_name = str(func.__name__)
+        qualified_name = str(func.__qualname__)
 
-        if setter is None and setter_name is None:
-            raise Exception("Either setter or setter_name needs to be defined")
+        if setter_name is None:
+            raise Exception("Setter_name needs to be defined")
 
         if min_value or max_value:
             descriptor = NumberSettingDescriptor(
-                id=str(property_name),
-                property=str(property_name),
+                id=qualified_name,
+                property=property_name,
                 name=name,
                 unit=unit,
-                setter=setter,
+                setter=None,
```

Comment:
It feels like this changes the behavior and breaks cases where the `setter` is used (like upcoming generic miot integration I'm working on).